zio-http/src/main/scala/zio/http/codec/HttpCodecType.scala

```diff
@@ -19,13 +19,15 @@ package zio.http.codec
 sealed trait HttpCodecType
 
 object HttpCodecType {
-  type RequestType <: Path with Content with Query with Header with Method
+  type RequestType <: Path with Content with Query with Header with Method with PathQuery
   type ResponseType <: Content with Header with Status
 
+  type PathQuery <: HttpCodecType
```

Comment:
Delete this, you do not need `PathQuery` (in addition to both `Query` and `Path`) and it makes reasoning about things very complex.

Reply:
Yes, I agree, introducing it can be mind bending to users who gets into the details of types.

However, the reason to introducing it is here:

https://github.com/zio/zio-http/pull/2154/files/d995976495ba0d24dcfc4d6108ccaecfbf556d5e#r1189264604